platform/src/EducationPlatformApp.js

```diff
@@ -895,7 +620,7 @@ class EducationPlatformApp {
             //actionRequestData.outputLanguage = outputLanguage;
 
         // Call backend conversion and service functions
-        let actionResultPromise = this.invokeActionFunction(buttonConfig.actionfunction, parameterMap);
+        let actionResultPromise = this.toolsManager.invokeActionFunction(buttonConfig.actionfunction, parameterMap);
```

Comment:
Not here, but slightly above: we call the `toolsManager` to get the actual function object. We only use this to determine the type of the `language` parameter. Is there way to refactor the code so that we don't have to access the function object here at all?

Reply:
Since the language type is always expected to be 'text' it could be set directly adding a constant `ACTION_FUNCTION_LANGUAGE_TYPE = "text"` which saves having to access it via `toolActionFunction`.

Quickly trying this out has also revealed that the OCL tool sets the language type to string so this would need to be update to use text to.

Reply:
Sounds like a sensible thing to do.

Reply:
Fixed in bcf669c333ac5b4bca5da6db8d202f322718398f and also requires change in PR thttps://github.com/mdenet/platformtools/pull/66 to be merged.